custom_components/adaptive_lighting/switch.py

```diff
@@ -1851,6 +1887,8 @@ async def _service_interceptor_turn_on_handler(
             adapt_brightness,
             adapt_color,
         )
+        if adaptation_data is None:
+            return
```

Comment:
FYI, a driving idea of the current design was simplicity, e.g., by specifically avoiding such checks (as well as the ones in `prepare_adaptation_data` and `_adapt_light`). If there's nothing to adapt, the adaptation data payload would be empty, the generator wouldn't return any actual service data objects, and no actual adaptation is carried out. (The processing overhead should be negligible.)

Reply:
This was nessesary for some earlier implementation I made, however, with the current one it is not required.

However, I was scratching my head why I was seeing `service_data`s with no color or brightness, so I think it doesn't hurt to keep it here.